fe/src/main/java/org/apache/doris/load/loadv2/Roaring64Map.java

```diff
@@ -0,0 +1,1484 @@
+            } else if (lowBitmap instanceof MutableRoaringBitmap) {
+                ((MutableRoaringBitmap) lowBitmap).flip(low);
+            } else {
+                // Fallback to a manual flip
+                if (lowBitmap.contains(low)) {
+                    lowBitmap.remove(low);
+                } else {
+                    lowBitmap.add(low);
+                }
+            }
+        }
+
+        invalidateAboveHigh(high);
+    }
+
+    /* ------------------ method below from Roaring64NavigableMap and being overwritten ----------------------------- */
+
+    /**
+     * Serialize this bitmap.
+     *
+     * Unlike RoaringBitmap, there is no specification for now: it may change from onve java version
+     * to another, and from one RoaringBitmap version to another.
+     *
+     * Consider calling {@link #runOptimize} before serialization to improve compression.
+     *
+     * The current bitmap is not modified.
+     *
+     * @param out the DataOutput stream
+     * @throws IOException Signals that an I/O exception has occurred.
+     */
+    public void serialize(DataOutput out) throws IOException {
+        if (highToBitmap.size() == 0) {
+            return;
+        }
+        if (is32BitsEnough()) {
+            out.write(BITMAP32);
+            highToBitmap.get(0).serialize(out);
+            return;
+        }
+
+        out.write(BITMAP64);
+        encodeVarint64(highToBitmap.size(), out);
+
+        for (Map.Entry<Integer, BitmapDataProvider> entry : highToBitmap.entrySet()) {
+            out.writeInt(entry.getKey().intValue());
+            entry.getValue().serialize(out);
+        }
+    }
+
+
+    /**
+     * Deserialize (retrieve) this bitmap.
+     *
+     * Unlike RoaringBitmap, there is no specification for now: it may change from one java version to
+     * another, and from one RoaringBitmap version to another.
+     *
+     * The current bitmap is overwritten.
+     *
+     * @param in the DataInput stream
+     * @throws IOException Signals that an I/O exception has occurred.
+     */
+    public void deserialize(DataInput in, int bitmapType) throws IOException {
+        this.clear();
+        highToBitmap = new TreeMap<>();
+
+        long nbHighs = 1;
+        if (bitmapType == BITMAP64) {
+            nbHighs = decodeVarint64(in);
+        }
+
+        for (int i = 0; i < nbHighs; i++) {
+            int high = in.readInt();
+            RoaringBitmap provider = new RoaringBitmap();
+            provider.deserialize(in);
+
+            highToBitmap.put(high, provider);
+        }
+
+        resetPerfHelpers();
+    }
+
+
+
+    /*---------------------------- method below is new written for doris's own bitmap --------------------------------*/
+
+    public boolean is32BitsEnough() {
+        return highToBitmap.size() == 1 && highToBitmap.get(0) != null;
+    }
+
+    public void add(long value) {
+        addLong(value);
+    }
+
+    public void add(int value) {
```

Comment:
```
    /**
     * Add value x
     *
     */
    void add(uint32_t x) {
        roarings[0].add(x);
        roarings[0].setCopyOnWrite(copyOnWrite);
    }
    void add(uint64_t x) {
        roarings[highBytes(x)].add(lowBytes(x));
        roarings[highBytes(x)].setCopyOnWrite(copyOnWrite);
    }
```
In BE.  container 0 content is 0 ~ std::numeric_limits<uint32_t>::max()

Reply:
Java doesn't support unsigned integer directly, must use additional method to transfer which means we need to judge whether a integer value fits in unsigned 32-bit manually
So I don't think there is a graceful implementation to support unsigned integer in java.
Java types is narrower than C++,So I think if we want to transfer bitmap between C++ and java,it's better to keep C++ compatible with java types actively.

Reply:
In Java，we could convert signed integer to unsigned integer easily.

There is no any reason to change C++ implementation, you should change java implementation.

Reply:
OK.
Now for Doris's  ```BitmapValue```,
java version support range [0, Long.MAX_VALUE]
C++ version support range [0,Long.MAX_VALUE * 2]

Reply:
OK